evap/evaluation/management/commands/anonymize.py

```diff
@@ -76,33 +72,41 @@ def anonymize_data(self):
             self.stdout.write("")
             raise
 
-    def randomize_text_answers(self, lorem_ipsum):
-        for text_answer in TextAnswer.objects.all():
-            text_answer.answer = self.lorem(text_answer.answer, lorem_ipsum)
-            if text_answer.original_answer:
-                text_answer.original_answer = self.lorem(text_answer.original_answer, lorem_ipsum)
-            text_answer.save()
 
-    @staticmethod
-    def lorem(text, lorem_ipsum):
-        word_count = len(text.split(' '))
-        return ' '.join(itertools.islice(itertools.cycle(lorem_ipsum), word_count))
+    # Anonymizes the email templates by replacing the actual subject with a sample one
+    # and replacing the body with lorem ipsum.
+    def anonymize_email_templates(self, lorem_ipsum):
+        self.stdout.write("TODO: Anonymize email templates...")
 
-    @staticmethod
-    def generate_random_usernames(first_names, last_names):
-        user_count = UserProfile.objects.count()
-        random_usernames = set()
-        while len(random_usernames) != user_count:
-            random_usernames.add((random.choice(first_names), random.choice(last_names)))
-        return random_usernames
 
-    @staticmethod
-    def replace_usernames_emails(random_usernames):
-        for user, new_username in zip(UserProfile.objects.all(), random_usernames):
+    # Replaces names, usernames, email addresses, login keys and valid until
+    # dates with fake ones.
+    def anonymize_users(self, first_names, last_names):
+        user_profiles = UserProfile.objects.all()
+
+        # Generate as many fake usernames as real ones exist. Use the provided
+        # first/last names for that.
+        self.stdout.write("Generating fake usernames...")
+        fake_usernames = set()
```

Comment:
@karyon have we ever thought about simplifying this? I don't see why we need this build-a-set-set-and-zip procedure, I think looping over all users and just giving him a random first and last name should do exactly the same, be faster, less code and simpler to understand.

Reply:
Ahh I see - no duplicates cause it's a set. Funny, I've thought about this, but somehow my mind did completely ignore the fact that sets can not have duplicates.